hyperspy/axes.py

```diff
@@ -825,6 +990,98 @@ def _on_scale_changed(self):
     def _on_offset_changed(self):
         self.events.any_axis_changed.trigger(obj=self)
 
+    def convert_units(self, axes=None, units=None, same_units=True,
+                      factor=0.25):
+        """ Convert the scale and the units of the selected axes. If the units
+        is not supported by the pint library, the scale and units are not
+        changed.
+
+        Parameters
+        ----------
+        axes : {int | string | iterable of `DataAxis` | None}
+            Default = None
+            Convert to the convenient scale and units on the specified axis.
+            If int, the axis can be specified using the index of the
+            axis in `axes_manager`.
+            If string, argument can be `navigation` or `signal` to select the
+            navigation or signal axes. The axis name can also be provided.
+            If `None`, convert all axes.
+        units : {list of string of the same length than axes | str | None}
+            Default = None
+            If list, the selected axes will be converted to the provided units.
+            If str, the navigation or signal axes will converted to the 
+            provided units.
+            If `None`, the scale and the units are converted to the appropriate 
+            scale and units to avoid displaying scalebar with >3 digits or too 
+            small number. This can be tweaked by the `factor` argument.
+        same_units : bool
+            If `True`, force to keep the same units if the units of
+            the axes differs. It only applies for the same kind of axis, 
+            `navigation` or `signal`. By default the converted units of the 
+            first axis is used for all axes. If `False`, convert all axes 
+            individually.
+        factor : float
+            Factor used to determine the prefix of the units.
+        """
+        convert_navigation = convert_signal = True
+
+        if axes is None:
+            axes = self.navigation_axes + self.signal_axes
+        elif axes == 'navigation':
+            axes = self.navigation_axes
+            convert_signal = False
+        elif axes == 'signal':
+            axes = self.signal_axes
+            convert_navigation = False
+        elif isinstance(axes, DataAxis) or type(axes) is int or type(axes) is str:
+            if not isinstance(axes, DataAxis):
+                axes = self[axes]
+            axes = (axes, )
+            convert_navigation = axes[0].navigate
+            convert_signal = not axes[0].navigate
+        else:
+            raise TypeError(
+                'Axes type `{}` is not correct.'.format(type(axes)))
+
+        if type(units) is str or units is None:
+            units = [units] * len(axes)
+        elif type(units) is list:
```

Comment:
`if isinstance(units, list) and len(units) != len(axes):`

i.e. combine the two conditionals

Reply:
Alternatively you could just loop the last one if not enough units are given:
```python
units = ['um', 'nm']
axes = [1,2,3]
units = units + units[-1:]*(len(axes) - len(units))
```

Reply:
Combining both conditionals is not working here, because it will skip the case when units is a list with the correct length and it will end up raising an error in the `else` block.

I am a bit sceptical about extrapolating the list of units, for the case when navigation and signals axis are passed. I would prefer to force the user to specify the correct units.